core/import-seac.c

```diff
@@ -0,0 +1,360 @@
+		//  0 = dive_number
+		//  1 = runtime_s
+		//  2 = depth_cm
+		//  3 = temperature_mCx10  - eg dC
+		//  4 = active_O2_fr
+		//  5 = first_stop_depth_cm
+		//  6 = first_stop_time_s
+		//  7 = ndl_tts_s
+		//  8 = cns
+
+	dive_start(state);
+	state->cur_dive->number = atoi(data[0]);
+
+	// Get time and date
+	sscanf(data[2],"%d/%d/%2d",&day,&month,&year);
+	sscanf(data[4],"%2d:%2d:%2d",&hour,&mins,&secs);
+
+	// TODO: Confirm Seac behaviour timezones
+	// Suspect a lookup table is used.
+	tz = atoi(data[3]);
+
+	// Timezone offset lookup array
+	const int timezoneoffset[][2] =
+	{{-12,   0},
+	 {-11,   0},
+	 {-10,   0},
+	 {-9 , -30},
+	 {-9 ,   0},
+	 {-8 ,   0},
+	 {-7 ,   0},
+	 {-6 ,   0},
+	 {-5 ,   0},
+	 {-4 , -30},
+	 {-4 ,   0},
+	 {-3 , -30},
+	 {-3 ,   0},
+	 {-2 ,   0},
+	 {-1 ,   0},
+	 { 0 ,   0},
+	 { 1 ,   0},
+	 { 2 ,   0},
+	 { 3 ,   0},
+	 { 3 ,  30},
+	 { 4 ,   0},
+	 { 4 ,  30},
+	 { 5 ,   0},
+	 { 5 ,  30},
+	 { 5 ,  45},
+	 { 6 ,   0},
+	 { 6 ,  30},
+	 { 7 ,   0},
+	 { 8 ,   0},
+	 { 8 ,  45},
+	 { 9 ,   0},
+	 { 9 ,  30},
+	 { 9 ,  45},
+	 { 10,   0},
+	 { 10,  30},
+	 { 11,   0},
+	 { 11,  30},
+	 { 12,   0},
+	 { 12,  45},
+	 { 13,   0},
+	 { 13,  45},
+	 { 14,   0}};
+
+	hour = hour + timezoneoffset[tz][0];
+	mins = mins + timezoneoffset[tz][1];
+
+	// Deal with minute over/underflow
+	if (mins > 59){
+		mins -= 60;
+		hour += 1;
+	} else if (mins < 0){
+		mins += 60;
+		hour -= 1;
+	}
+
+	// Handle hour overflow
+	if (hour < 0){
+		hour += 24;
+		day -= 1;
+	} else if (hour > 23){
+		hour -= 24;
+		day += 1;
+	}
+
+	// Year is scanned in as two digits, so add 2000
+	year += 2000;
+	// Leap year check
+	if (!(year % 400)){
+	    isleap = true;
+	} else if (!(year % 100)){
+		isleap = false;
+	} else if (!(year % 4)){
+		isleap = true;
+	}
+
+	// Handle Day overflow
+	if (day > 31){
+		day -= 31;
+		month += 1;
+	// remember, months are as scanned in here
+	} else if (day > 30 && (month == 4 || month == 6 || month == 9 || month == 11)){
+		day -= 30;
+		month += 1;
+	} else if (day > 28 && month == 2 && isleap == false){
+		day -= 28;
+		month += 1;
+	} else if (day > 29 && month == 2 && isleap == true){
+		day -= 29;
+		month += 1;
+	}
+
+
+	// Handle month overflow
+	if (month > 12){
+		year += 1;
+		month -= 12;
+	}
+
+	divetime.tm_year = year - 1900;
+	divetime.tm_mon = month - 1;
+	divetime.tm_mday = day;
+	divetime.tm_hour = hour;
+	divetime.tm_min = mins;
+	// WARNING: Possibly Non portable
+	divetime.tm_sec = secs - timezone; // counteract mktime() assumption of local time
+	tmdivetime = mktime(&divetime);
+
+	if (tmdivetime == -1){
+		fprintf(stderr,"Failed to parse dive date and time during import");
+	} else {
+		state->cur_dive->when = tmdivetime;
+	}
+
+	// 6 = dive_type
+	// TODO: confirm freedive recorded as type 2
+	if (data[6]) {
+		switch (atoi(data[6])){
+		case 1:
+			state->cur_dive->dc.divemode = OC;
+			break;
+		case 2:
+			state->cur_dive->dc.divemode = FREEDIVE;
+			break;
+		}
```

Comment:
I wonder if there should be a default clause to print the unknown value when verbose mode is enabled. This might make it easier to add divemodes if we receive a user report certain mode is missing.

Reply:
Good idea for the default case. Added some output to stderr if verbose flag is set.